Foundation/include/Poco/NumberFormatter.h

```diff
@@ -59,20 +68,36 @@ class Foundation_API NumberFormatter
 		/// right justified and zero-padded in a field
 		/// having at least the specified width.
 
-	static std::string formatHex(int value, bool prefix = false);
+	[[deprecated("use formatHex with options instead")]]
+	static std::string formatHex(int value, bool prefix);
 		/// Formats an int value in hexadecimal notation.
 		/// If prefix is true, "0x" prefix is prepended to the
 		/// resulting string.
 		/// The value is treated as unsigned.
 
-	static std::string formatHex(int value, int width, bool prefix = false);
+	static std::string formatHex(int value, Options options = Options::OPT_NONE);
+		/// Formats an int value in hexadecimal notation.
+		/// Options (see NumberFormatter::Options) define the format of the
+		/// generated string.
+		/// The value is treated as unsigned.
+
+	[[deprecated("use formatHex with options instead")]]
+	static std::string formatHex(int value, int width, bool prefix);
```

Comment:
So why do you remove default prefix false? That change will break things if someone is using these functions.

Reply:
Aa I see it now. That was probably issue in first place. Because those are ambiguous with wrong usage.

Reply:
I agree - deprecated functions' reason for existence is precisely not to break code and give people time to adjust

Reply:
I removed the default prefix because there is ambiguity and compile fails.

```
[[deprecated("use formatHex with options instead")]]
static std::string formatHex(int value, int width, bool prefix = false);
static std::string formatHex(int value, int width, Options options = Options::OPT_NONE);

```
Default was no prefix which is also the default when using options. The behaviour for the user if it did not use the `prefix` does not change. Deprecation warning is issues only for users that explicitly used bool prefix argument.

```
/Users/matejk/git/github/poco/Foundation/src/Logger.cpp:252:18: error: call to 'formatHex' is ambiguous
                message.append(NumberFormatter::formatHex(addr, 4));
                               ^~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/matejk/git/github/poco/Foundation/include/Poco/NumberFormatter.h:704:37: note: candidate function
inline std::string NumberFormatter::formatHex(int value, int width, bool prefix)
                                    ^
/Users/matejk/git/github/poco/Foundation/include/Poco/NumberFormatter.h:710:37: note: candidate function
inline std::string NumberFormatter::formatHex(int value, int width, Options options)
                                    ^
```

Reply:
Aa true. Nice